projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -24,63 +39,137 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   ];
   prevSortMeta = "-at member_id";
 
-  @Output() onAccept = new EventEmitter<any>();
-  @Output() onReject = new EventEmitter<any>();
+  acceptLoading = false;
+  rejectLoading = false;
+  requestLoading = false;
+  requestAction = "";
+  selection = [];
 
-  groupSwitch = [
-    {
-      label: "This group only",
-    },
-    {
-      label: "All subgroups",
-    },
-  ];
+  _setRequestData(sortBy = GROUP_REQUESTS_API.sort) {
```

Comment:
This function changes completely its role. `_setRequestData` is definitely not appropriate. (maybe "reloadData" would?)

Also, I haven't check into details yet, but this sort attribute in param is suspicious, it would mean some kind of reload would not care about the defined sort? (

Reply:
That destroys the defined sort, I need to save the sort info.

Reply:
fixed
This was the previous sort.
![image](https://user-images.githubusercontent.com/51055029/82228042-2cbed900-995b-11ea-87a0-f1af289e86c3.png)
Current sort after fixing.
![image](https://user-images.githubusercontent.com/51055029/82228813-34cb4880-995c-11ea-9241-38da33bc602c.png)
It keeps the previous sort info.